### PR TITLE
Simplified InternalConnectionStream

### PR DESCRIPTION
**Pull request by rozza**
Now assumes exclusive use of the stream. The plan is for multithread use again in the future.
JAVA-1674

[Review on Reviewable](https://reviewable.io/reviews/rozza/mongo-java-driver/72)

### CODE REVIEW

File: driver-core/src/main/com/mongodb/connection/InternalStreamConnection.java [hunk 1]
```diff
@@ -203,61 +186,84 @@ public ResponseBuffers receiveMessage(final int responseTo) {
...
+                    new MongoSocketClosedException("Cannot read from a closed stream", getServerAddress()));
+        } else {
+            fillAndFlipBuffer(REPLY_HEADER_LENGTH,
+                    errorHandlingCallback(new ResponseHeaderCallback(new SingleResultCallback<ResponseBuffers>() {
+                        @Override
+                        public void onResult(final ResponseBuffers result, final Throwable t) {
+                            if (LOGGER.isTraceEnabled()) {
+                                LOGGER.trace(format("Received message: %s", responseTo));
+                            }
+                            callback.onResult(result, t);
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/connection/InternalStreamConnection.java]: Did we lose the errorHandlingCallback here?

Comment (rozza) [inline driver-core/src/main/com/mongodb/connection/InternalStreamConnection.java]: Yes - fixed

**Comment by jyemin** [comment]
Looks good.  

Please also update the Javadoc for com.mongodb.connection.Connection to indicate that Connection may not be thread safe, so you have to do your own synchronization.

**Comment by rozza** [comment]
Done

**Comment by rozza** [comment]
Already merged

**Pull request closed without merging**